packages/react-core/src/next/components/Select/Select.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { css } from '@patternfly/react-styles';
+import { Menu, MenuContent, MenuProps } from '../../../components/Menu';
+import { Popper } from '../../../helpers/Popper/Popper';
+
+export interface SelectProps extends MenuProps {
+  /** Anything which can be rendered in a select */
+  children?: React.ReactNode;
+  /** Classes applied to root element of select */
+  className?: string;
+  /** Flag to indicate if select is open */
+  isOpen?: boolean;
+  /** Single itemId for single select menus, or array of itemIds for multi select. You can also specify isSelected on the MenuItem. */
+  selected?: any | any[];
+  /** Renderer for a custom select toggle. Forwards a ref to the toggle. */
+  toggle: (toggleRef: React.RefObject<any>) => React.ReactNode;
+  /** Function callback called when user selects item. */
+  onSelect?: (event?: React.MouseEvent<Element, MouseEvent>, itemId?: string | number) => void;
+  /** Callback to allow the dropdown component to change the open state of the menu.
+   * Triggered by clicking outside of the menu, or by pressing either tab or escape. */
+  onOpenChange?: (isOpen: boolean) => void;
+  /** Indicates if the select should be without the outer box-shadow */
+  isPlain?: boolean;
+  /** Min width of the select menu */
+  minWidth?: string;
+  /** @hide Forwarded ref */
+  innerRef?: React.Ref<HTMLDivElement>;
+}
+
+const SelectBase: React.FunctionComponent<SelectProps> = ({
+  children,
+  className,
+  onSelect,
+  isOpen,
+  selected,
+  toggle,
+  onOpenChange,
+  isPlain,
+  minWidth,
+  innerRef,
+  ...props
+}: SelectProps) => {
+  const localMenuRef = React.useRef<HTMLDivElement>();
+  const toggleRef = React.useRef<HTMLButtonElement>();
+  const containerRef = React.useRef<HTMLDivElement>();
+
+  const menuRef = (innerRef as React.RefObject<HTMLDivElement>) || localMenuRef;
+  React.useEffect(() => {
+    const handleMenuKeys = (event: KeyboardEvent) => {
+      if (!isOpen && toggleRef.current?.contains(event.target as Node)) {
+        // toggle was clicked open, focus on first menu item
+        if (event.key === 'Enter') {
+          setTimeout(() => {
```

Comment:
Yeah I tried to add support for space in, but it seems I'm running into the same issue you described.

I also removed the extra handling from the toggle clicks. I had forgotten that had moved internally when I went back to create the examples.

Reply:
Yeah I imagine it might take a little more investigating and I don't think it's necessarily worth holding this PR up over (especially since this is beta anyway)

Reply:
I'll open a follow up to investigate space